File: android_feature_bookmark_edit/src/main/java/com/app/feature/bookmark/edit/EditBookmarkViewState.kt
package com.app.feature.bookmark.edit

import com.app.shared.business.EditBookmarkState
import com.app.views.viewstate.TopicViewState

data class EditBookmarkViewState(val state: EditBookmarkState) {

    val bookmarkId = state.bookmark.id

    val topicsViewState: List<TopicViewState.Selectable> = state.topics.map {
        TopicViewState.Selectable(state = it, isSelected = it.id == state.bookmark.topic?.id)
    }
}

File: ios_app/ios_feature_bookmark_edit/ios_feature_bookmark_edit/EditBookmarkViewState.swift
//
//  EditBookmarkViewState.swift
//  ios_feature_bookmark_edit
//
//  Created by Gabriel Coman on 02/05/2020.
//  Copyright © 2020 Gabriel Coman. All rights reserved.
//

import Foundation
import RememberShared
import ios_views

struct EditBookmarkViewState {
    
    private let state: EditBookmarkState?
    
    init(state: EditBookmarkState? = nil) {
        self.state = state
    }
    
    var bookmarkId: Int32 {
        return state?.bookmark.id ?? 0
    }
    
    var viewStates: [TopicViewState.Selectable] {
        return state?.topics.map {
            TopicViewState.Selectable(
                state: $0,
                isSelected: $0.id == state?.bookmark.topic?.id,
                noBookmarks: 0
            )
        } ?? []
    }
}